v2/pkg/operators/common/dsl/dsl.go

```diff
@@ -915,6 +917,38 @@ func init() {
 
 			return buf.String(), nil
 		}),
+		"resolve": makeDslWithOptionalArgsFunction("(host string, format string) string", func(args ...interface{}) (interface{}, error) {
```

Comment:
As mentioned in https://github.com/projectdiscovery/nuclei/pull/3321#issuecomment-1429052031 I think it would be a great addition to be able to specify the DNS query type. In my opinion, since the second argument is a string, this can be easily accomplished with a switch statement that maps `format` to `dns.TypeXXX`, for example:
```go
switch format {
case "4","a":
dns.TypeA
case "6","aaaa":
dns.TypeAAAA
case "ptr":
dns.TypePTR
...
}
```